### PR TITLE
conformance uploader: dry run and glob junit

### PR DESCRIPTION
**Pull request by BenTheElder**
- adds optional `--dry-run` flag to disable actually uploading anything, handy for debugging
- updates the `--junit` flag to use [`glob.glob`](https://docs.python.org/2/library/glob.html) which is ~ backwards compatible but also allows specifying a path like `--junit=/foo/bar/junit_*.xml` to upload all junit.

/area testgrid

### CODE REVIEW

File: testgrid/conformance/upload_e2e.py [hunk 1]
```diff
@@ -180,19 +185,32 @@ def parse_args(cli_args=None):
...
+        help='if set, do not actually upload anything, only print actions',
+        required=False,
+        action='store_true',
+    )
     return parser.parse_args(args=cli_args)
 
 def main(cli_args):
     args = parse_args(cli_args)
-    log, junit, year, bucket = args.log, args.junit, args.year, args.bucket
+    log, year, bucket, dry = args.log, args.year, args.bucket, args.dry_run
```
[the remark this answers is not in the document]
Comment (BenTheElder) [reply testgrid/conformance/upload_e2e.py]: The command line flag is `--dry-run` but will be stored in the parsed args object as `dry_run` by python's argument parser library.

File: testgrid/conformance/upload_e2e.py [hunk 2]
```diff
@@ -207,11 +225,11 @@ def main(cli_args):
...
-    upload_string(gcs_dir+'/started.json', started_json)
-    upload_string(gcs_dir+'/finished.json', finished_json)
-    upload_file(gcs_dir+'/build-log.txt', log)
-    upload_file(gcs_dir+'/artifacts/'+path.basename(junit), junit)
-
+    upload_string(gcs_dir+'/started.json', started_json, dry)
+    upload_string(gcs_dir+'/finished.json', finished_json, dry)
+    upload_file(gcs_dir+'/build-log.txt', log, dry)
+    for junit_file in junits:
+        upload_file(gcs_dir+'/artifacts/'+path.basename(junit_file), junit_file, dry)
```
Comment (AishSundar) [inline testgrid/conformance/upload_e2e.py]: So per this logic we will upload 'n' junit files to artifacts/ in GCS bucket. I am assuming TG supports handling multiple junits?

**Comment by BenTheElder** [comment]
Yes, testgrid supports this. Eg:
http://gcsweb.k8s.io/gcs/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce/25509/artifacts/

https://k8s-testgrid.appspot.com/sig-release-master-blocking#gci-gce

**Pull request merged**